serve/publish.go

```diff
@@ -0,0 +1,180 @@
+	PackageTypeDocker PackageType = "docker"
+)
+
+// manifest is the plugin.json file inside the dist directory. It is used by CloudQuery registry
+// to be able to publish the plugin with all the needed metadata.
+type Manifest struct {
+	Name             string               `json:"name"`
+	Version          string               `json:"version"`
+	Title            string               `json:"title"`
+	ShortDescription string               `json:"short_description"`
+	Description      string               `json:"description"`
+	Categories       []string             `json:"categories"`
+	Protocols        []int                `json:"protocols"`
+	SupportedTargets []plugin.BuildTarget `json:"supported_targets"`
+	PackageType      PackageType          `json:"package_type"`
+}
+
+func isDirectoryExist(path string) bool {
+	info, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return info.IsDir()
+}
+
+func (s *PluginServe) writeTablesJson(ctx context.Context, dir string) error {
+	tables, err := s.plugin.Tables(ctx, plugin.TableOptions{
+		Tables: []string{"*"},
+	})
+	if err != nil {
+		return err
+	}
+	buffer := &bytes.Buffer{}
+	m := json.NewEncoder(buffer)
+	m.SetIndent("", "  ")
+	m.SetEscapeHTML(false)
+	err = m.Encode(tables)
+	if err != nil {
+		return err
+	}
+	outputPath := filepath.Join(dir, "tables.json")
+	return os.WriteFile(outputPath, buffer.Bytes(), 0644)
+}
+
+func (p *PluginServe) build(ctx context.Context, pluginDirectory string, goos string, goarch string) error {
+	pluginName := "plugin" + "_" + goos + "_" + goarch
+	distPath := pluginDirectory + "/dist"
+
+	pluginPath := distPath + "/" + pluginName
+	args := []string{"build", "-C", pluginDirectory, "-o", pluginPath}
+	cmd := exec.Command("go", args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Env = os.Environ()
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to build plugin with `go %v`: %w", args, err)
+	}
+
+	pluginFile, err := os.Open(pluginPath)
+	if err != nil {
+		return fmt.Errorf("failed to open plugin file: %w", err)
+	}
+	defer pluginFile.Close()
+
+	zipPluginPath := pluginPath + ".zip"
+	zipPluginFile, err := os.Create(zipPluginPath)
+	if err != nil {
+		return fmt.Errorf("failed to create zip file: %w", err)
+	}
+	defer zipPluginFile.Close()
+
+	zipWriter := zip.NewWriter(zipPluginFile)
+	defer zipWriter.Close()
+
+	pluginZip, err := zipWriter.Create(pluginName)
+	if err != nil {
+		return fmt.Errorf("failed to create file in zip archive: %w", err)
+	}
+	if err := os.Remove(pluginPath); err != nil {
+		return fmt.Errorf("failed to remove plugin file: %w", err)
+	}
+	_, err = io.Copy(pluginZip, pluginFile)
+	if err != nil {
+		return fmt.Errorf("failed to copy plugin file to zip archive: %w", err)
+	}
+	return nil
+}
+
+func (s *PluginServe) writeManifest(ctx context.Context, dir string) error {
+	manifest := Manifest{
+		Name:             s.plugin.Name(),
+		Version:          s.plugin.Version(),
```

Comment:
I think we're missing the part of injecting the plugin version via Go Releaser here https://github.com/cloudquery/cloudquery/blob/db142bc867e631f00cb06f21a4a0566210fc7fd0/plugins/.goreleaser.yaml#L12

Not sure how this would work in this approach. We could configure release please maybe to update the version in the `go` code like we do with python, see https://github.com/cloudquery/cloudquery/pull/12869/files#diff-e10085e6cd58ab584a2a0fd98e5b4bcdda801327fe2f8e571ee049a3bceaf1e0L13